daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala

```diff
@@ -235,13 +335,349 @@ class EvaluationOrderTest extends AnyFreeSpec with Matchers with Inside {
+                  "",
+                ),
+              )
+            ),
+          )
+
+          inside(res) {
+            case Success(Left(SErrorDamlException(IE.WronglyTypedContract(_, T, Dummy)))) =>
+          }
+          msgs shouldBe Seq.empty
+        }
+
+        // TEST_EVIDENCE: Semantics: Evaluation order of fetch of global contract with failure authorization
+        "authorization failures" in {
+          val (res, msgs) = evalUpdateApp(
+            pkgs,
+            e"""\(cId: ContractId M:T) -> fetch @M:T cId""",
+            Array(SContractId(cId)),
+            charlie,
+            getContract = getContract,
+          )
+
+          inside(res) { case Success(Left(SErrorDamlException(IE.FailedAuthorization(_, _)))) => }
+          msgs shouldBe globalFetchByIdTraces
+        }
+      }
+
+      "a local contract" - {
+
+        // TEST_EVIDENCE: Semantics: Evaluation order of successful fetch of a local contract
+        "successful fetch" in {
+
+          val (res, msgs) = evalUpdateApp(
+            pkgs,
+            e"""\(sig : Party) (obs : Party) ->
+             ubind cId: ContractId M:T <- create @M:T M:T { signatory = sig, observer = obs, precondition = True, key = M:toKey sig, nested = M:buildNested 0 }
+             in fetch @M:T cId""",
+            Array(SParty(alice), SParty(bob)),
+            alice,
+          )
+
+          inside(res) { case Success(Right(_)) => }
+          msgs shouldBe createTraces :++ localFetchByIdTraces
+        }
+
+        // TEST_EVIDENCE: Semantics: Evaluation order of fetch of an inactive local contract
+        "inactive contract" in {
+
+          val (res, msgs) = evalUpdateApp(
+            pkgs,
+            e"""\(sig : Party) (obs : Party) ->
+             ubind 
+               cId: ContractId M:T <- create @M:T M:T { signatory = sig, observer = obs, precondition = True, key = M:toKey sig, nested = M:buildNested 0 } ;
+               x: Unit <- exercise @M:T Archive cId sig
+             in fetch @M:T cId""",
+            Array(SParty(alice), SParty(bob)),
+            alice,
+          )
+
+          inside(res) { case Success(Left(SErrorDamlException(IE.ContractNotActive(_, T, _)))) => }
+          msgs shouldBe createTraces :+ "archive"
+        }
+
+        // TEST_EVIDENCE: Semantics: Evaluation order of fetch of an wrongly typed local contract
+        "wrongly typed contract" in {
+
+          val (res, msgs) = evalUpdateApp(
+            pkgs,
+            e"""\(sig : Party) ->
+             ubind cId1: ContractId M:Dummy <- create @M:Dummy M:Dummy { signatory = sig } 
+             in let cId2: ContractId M:T = COERCE_CONTRACT_ID @M:Dummy @M:T cId1
+             in fetch @M:T cId2""",
+            Array(SParty(alice)),
+            alice,
+          )
+
+          inside(res) {
+            case Success(Left(SErrorDamlException(IE.WronglyTypedContract(_, T, Dummy)))) =>
+          }
+          msgs shouldBe Seq.empty
+        }
```

Comment:
I assume it is not possible to have an authorization failure for a local contract

Reply:
That’s not correct, the authorization context can change throughout a transaction. See https://gist.github.com/cocreature/e3257ce5e0855a4e0e6f7b8ea148d926 for an example.